app/models/settings.rb

```diff
@@ -118,4 +121,11 @@ def update_approval_required(attrs)
   def sanitize_attributes(attrs)
     attrs.reject { |key, value| self.class.non_null_columns_names.include?(key.to_s) && value.to_s.empty? }
   end
+
+  def enforce_sso_allowed?
+    enforce_sso = EnforceSSOValidator.new(account: account)
+    return if enforce_sso.valid?
+
+    errors.add(:enforce_sso, enforce_sso.error_message)
```

Comment:
another nitpick. Using a variable with name same as an attribute can be confusing.
```suggestion
    sso_validator = EnforceSSOValidator.new(account: account)
    return if sso_validator.valid?

    errors.add(:enforce_sso, sso_validator.error_message)
```

Reply:
Yes, 100%! I didn't realize...

Reply:
I think you updated the controller while my concern was more about the model. Perhaps not intentional.

Reply:
Agree with @akostadinov here.